### ISSUE TITLE
Confusing help message on an attempt to call a module-local macro

### ISSUE THREAD

**Issue by nodakai** [rust-lang/rust#31660]
```rust
mod m {
    macro_rules! k {
        () => ()
    }
}

fn main() {
    k!();
}
```

```
<anon>:8:5: 8:6 error: macro undefined: 'k!'
<anon>:8     k!();
             ^
<anon>:8:5: 8:6 help: did you mean `k!`?
```
The help message is off the point.

- https://play.rust-lang.org/?gist=f707c254c6feded72200&version=nightly

**Comment by GuillaumeGomez** [comment]
I'm looking at it. The function in cause is [find_best_match_for_name](https://github.com/rust-lang/rust/blob/master/src/libsyntax/util/lev_distance.rs#L49).

**Comment by nodakai** [comment]
Already fixed by #31707, closing
